pkg/chattr/chattr.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2025 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package chattr
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+const (
+	// Attribute flags
+	FS_IMMUTABLE_FL = 0x00000010 // Immutable file
+	FS_APPEND_FL    = 0x00000020 // Append-only file
+	// ... other flags as needed
+)
+
+// SetAttr sets the attributes of a file.
+func SetAttr(file *os.File, attrStr string) error {
+	var attr int32
```

Comment:
Datatype should match the type in the kernel: https://elixir.bootlin.com/linux/v4.5/source/include/uapi/linux/fs.h#L238